server/container_list.go

```diff
@@ -28,6 +28,46 @@ func filterContainer(c *pb.Container, filter *pb.ContainerFilter) bool {
 	return true
 }
 
+// filterContainerList applies a protobuf-defined filter to retrieve only intended containers. Not matching
+// the filter is not considered an error but will return an empty response.
+func (s *Server) filterContainerList(filter *pb.ContainerFilter, origCtrList []*oci.Container) (ctrList []*oci.Container, err error) {
+	ctrList = origCtrList
+
+	// Filter using container id and pod id first.
+	if filter.Id != "" {
+		id, err := s.CtrIDIndex().Get(filter.Id)
+		if err != nil {
+			// If we don't find a container ID with a filter, it should not
+			// be considered an error.  Log a warning and return an empty struct
+			logrus.Warn("unable to find container ID %s", filter.Id)
+			return []*oci.Container{}, nil
+		}
+		c := s.ContainerServer.GetContainer(id)
+		if c != nil {
+			if filter.PodSandboxId != "" {
+				if c.Sandbox() == filter.PodSandboxId {
+					ctrList = []*oci.Container{c}
+				} else {
+					ctrList = []*oci.Container{}
+				}
+
+			} else {
+				ctrList = []*oci.Container{c}
+			}
+		}
+	} else {
+		if filter.PodSandboxId != "" {
+			pod := s.ContainerServer.GetSandbox(filter.PodSandboxId)
+			if pod == nil {
+				ctrList = []*oci.Container{}
+			} else {
+				ctrList = pod.Containers().List()
+			}
+		}
+	}
+	return
```

Comment:
return origCtrList

Reply:
Thanks, changes made (and its much less spaghetti-like)

Reply:
Still got a lint error.